### PR TITLE
Awesome books: refactor to use JavaScript classes

### PR DESCRIPTION
**Pull request by lfmnovaes**
[Project Requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/books/m2_plain_js_classes.md)
---
- Add the addBook inside the Book class.
- Add the removeBook inside the Book class.
- OnLoad, Add and Remove triggers refresh function, which syncs the array of objects with the localStorage.
- Changed the logic based on ESLint suggestions

### CODE REVIEW

File: js/scripts.js [hunk 1]
```diff
@@ -1,9 +1,4 @@
-class Book {
-  constructor(title, author) {
-    this.title = title;
-    this.author = author;
-  }
-}
+let books = [];
```
Comment (Lameck1) [inline js/scripts.js]: - [x] We are allowed to use Arrays but only inside a class. Remember we are refactoring our project to use classes that will hold data and provide us with methods that will help us manipulate the data(_the books array in this case_). See the screenshot below for reference:

| Requirement |
| --- |
| useClasses |

Consider refactoring this array to the Book class as a static property. For example:

```JavaScript
class Book {
  static books = [];
  
  // if you would like to access this static property, start with the name of the class as below
  Book.books.push({ title: this.title, author: this.author });
}
```

Alternatively, you can have two classes, one that defines a book object and another that acts as a BookStore. In this case, the BookStore class will be the one to have the methods for adding and removing books.

File: js/scripts.js [hunk 2]
```diff
@@ -18,47 +13,55 @@ const addLi = (title, author) => {
...
+  removeBook(e) {
+    const tgt = e.target;
+    if (tgt.tagName === 'BUTTON') {
+      [this.title] = e.composedPath()[1].firstChild.data.split(' - ');
+      books = books.filter((el) => el.title !== this.title);
+      tgt.closest('li').remove();
+    }
+    refresh();
+  }
+}
```
Comment (Lameck1) [inline js/scripts.js]: - [x] We are failing in one of the requirements as concerns deleting a book. Look at the screenshots below for reference.

**Firstly, a screenshot from the previous task, for emphasis**
| --- |
| earlier_requirement |

**Secondly, the specific requirement in this task that requires us to maintain the functionality**
| --- |
| new_requirement |

## The problem

Right now, the app allows the addition of duplicate books(**_books with the same title_**), and when the user tries to remove just one, all the books with the same title will all be deleted. We are required to delete only one book at a time.

## Remedy

Let the app only accept unique books. OR, add a unique ID field to a Book object, so we may filter the books array with id property which is unique and not the title property.

**Review by Lameck1 — CHANGES_REQUESTED** [review]
Hi @lfmnovaes & @alvarorf, 

Awesome job refactoring the code to use JS class implementation 👏🏼!
There are just but some minor issues making us not move ahead. Let's kindly work on them.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: js/scripts.js (hunk 2 above)
Comment (lfmnovaes) [reply js/scripts.js]: https://github.com/lfmnovaes/awesomebooks/blob/8f451839572a931aa62d3229c54ec15a14993c34/js/scripts.js#L5
We chose to create a unique ID for each book created and deleting using it.

File: js/scripts.js (hunk 1 above)
Comment (lfmnovaes) [reply js/scripts.js]: Created a Book class and a Bookshelf class to solve this.

**Review by arslanbisharat — APPROVED** [review]
# Approved 👏 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of code reviews per project which you can see from your dashbaord. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

**Pull request merged**